cmd/archive_test.go

```diff
@@ -128,6 +128,110 @@ func TestArchiveContainsEnv(t *testing.T) {
 	require.Equal(t, "ipsum", metadata.Env["ENV2"])
 }
 
+func TestArchiveContainsLegacyCloudSettings(t *testing.T) {
+	t.Parallel()
+
+	// given a script with the cloud options
+	fileName := "script.js"
+	testScript := []byte(`
+		export let options = {
+			ext: {
+				loadimpact: {
+					distribution: {
+						one: { loadZone: 'amazon:us:ashburn', percent: 30 },
+						two: { loadZone: 'amazon:ie:dublin', percent: 70 },
+					},
+				},
+			},
+		};
+		export default function () {}
+	`)
+	ts := tests.NewGlobalTestState(t)
+	require.NoError(t, fsext.WriteFile(ts.FS, filepath.Join(ts.Cwd, fileName), testScript, 0o644))
+
+	// when we do archiving
+	ts.CmdArgs = []string{"k6", "archive", fileName}
+
+	newRootCommand(ts.GlobalState).execute()
+	require.NoError(t, untar(t, ts.FS, "archive.tar", "tmp/"))
```

Comment:
Can we also add a test with a static archive, please? In this way, we don't directly depend on the _current_ logic of the archive command. So, in the event, we will change the logic (for whatever reason) then we are still sure that an old archive is compatible.

Reply:
Consider a script that includes the following options:

```js
  ext: {
    loadimpact: {
      name: 'Hello k6 cloud!',
      distribution: {
        distributionLabel1: { loadZone: 'amazon:us:ashburn', percent: 50 },
        distributionLabel2: { loadZone: 'amazon:ie:dublin', percent: 50 },
      },
      note: 'Anything that may be worth noting about your test.',
   },
```

If we archive and we reuse it for running a script in the Cloud then the options are used for defining the distribution.

```
k6 archive script.js
k6 cloud archive.tar
```
So, if we break the archive regarding the Cloud part I'm not sure it will fail somewhere if we don't add one here.

Reply:
Okay, sounds like a  not common, but still valid case, so I will add the test :+1:

Reply:
Extracted as separated PR https://github.com/grafana/k6/pull/3402